pkg/controller/deployment/util/deployment_util_test.go

```diff
@@ -462,19 +462,24 @@ func TestEqualIgnoreHash(t *testing.T) {
 
 	for _, test := range tests {
 		equal, err := equalIgnoreHash(test.former, test.latter)
-		if err != nil {
-			t.Errorf("In test case %q, returned unexpected error %v", test.test, err)
-		}
-		if equal != test.expected {
-			t.Errorf("In test case %q, expected %v", test.test, test.expected)
-		}
-		equal, err = equalIgnoreHash(test.latter, test.former)
-		if err != nil {
-			t.Errorf("In test case %q (reverse order), returned unexpected error %v", test.test, err)
-		}
-		if equal != test.expected {
-			t.Errorf("In test case %q (reverse order), expected %v", test.test, test.expected)
+		runTest := func(t1, t2 api.PodTemplateSpec, reversed bool) {
+			reverseString := ""
+			if reversed {
+				reverseString = " (reverse order)"
+			}
+			if err != nil {
+				t.Errorf("In test case %q%s, expected no error, returned %v", test.test, reverseString, err)
+			}
+			if equal != test.expected {
+				t.Errorf("In test case %q%s, expected %v", test.test, reverseString, test.expected)
+			}
+			if t1.Labels == nil || t2.Labels == nil {
+				t.Errorf("In test case %q%s, unexpected labels becomes nil", test.test, reverseString)
+			}
 		}
+		runTest(test.former, test.latter, false)
+		// Test the same case in reverse order
+		runTest(test.latter, test.former, true)
 	}
```

Comment:
Add a test that does a deep copy of the input, then compares against the input again after calling equalIgnoreHash, to ensure that equalIgnoreHash doesn't modify its inputs?